src/Controller/Controller.php

```diff
@@ -544,6 +552,57 @@ public function invokeAction(Closure $action, array $args): void
         }
     }
 
+    /**
+     * Register middleware for the controller.
+     *
+     * @param string|\Closure|\Psr\Http\Server\MiddlewareInterface $middleware Middleware.
+     * @param array $options Valid options:
+     *  - `only`: (array|string) Only run the middleware for specified actions.
+     *  - `except`: (array|string) Run the middleware for all actions except the specified ones.
+     * @return void
+     * @psalm-var array{?only: string|array, ?except: string|array}
+     */
+    public function middleware($middleware, array $options = [])
+    {
+        $this->middlewares[] = [
+            'middleware' => $middleware,
+            'options' => $options,
+        ];
+    }
+
+    /**
+     * Get middleware to be applied for this controller.
+     *
+     * @return array
+     */
+    public function getMiddleware(): array
```

Comment:
```suggestion
    public function getMiddlewares(): array
```

This should be plural, right?

Reply:
Yes, but I was going for consistency since `RouteCollection`, `RouteBuilder` and `Route` also have methods of the same name which return arrays :slightly_frowning_face:.

Reply:
Should the property also be `middleware` then?

Reply:
Fair enough.